src/core_processor/camera_calibration/charuco_board_detection/charuco_board_definition.py

```diff
@@ -0,0 +1,10 @@
+import cv2
+
+aruco_marker_dict = cv2.aruco.Dictionary_get(cv2.aruco.DICT_4X4_250)
+number_of_squares_width = 7
+number_of_squares_height = 5
+black_square_side_length = 1
+aruco_marker_length_proportional = .8
+
+charuco_board_object = cv2.aruco.CharucoBoard_create(number_of_squares_width, number_of_squares_height, black_square_side_length, aruco_marker_length_proportional, aruco_marker_dict)
+number_of_charuco_corners = (number_of_squares_width - 1) * (number_of_squares_height - 1)
```

Comment:
this would be a wonderful class.

Reply:
indeed it would! 

...it'd be even more wonderful if I could figure out how to make it `frozen`, but I don't know how to do that while also generating the `charuco_board_object` on the basis of the class properties.

Reply:
you can make another class as a dataclass with fields that are frozen. Does that hit what you're looking for?

Reply:
oh yeah, totally! Thanks!